tools/draw-importances.py

```
@@ -1,55 +1,180 @@
+    Mode 2: Batch file processing for multiple hres and imps data\n
+    Output files will be saved in out/YYYYMMDD_HH folder in -imps.png files.
 
-frame = Image.open(sys.argv[1])
-frame = frame.resize((frame.width * frame_size_multiplier, frame.height * frame_size_multiplier))
-frame = frame.convert(mode='RGB')
+    """
+    if verbose:
+        click.echo("Input given: "+ str(input))
+        click.echo("Path given: " + str(path))
 
-mv_original_block_size = 8 // 2 # The importances are in 8×8 blocks, but we use half-resolution images.
-mv_block_size = mv_original_block_size * frame_size_multiplier
+    rx = re.compile(r'(\d\d-imps)|(\d-imps)')
 
-draw = ImageDraw.Draw(frame, mode='RGBA')
+    if input != (None, None):
+        if verbose:
+            click.secho("MODE 1", fg="yellow")
+        with open(input[1], 'rb') as f:
+            contents = f.read()
 
-# Draw the grid.
-for i in range(0, frame.width, mv_block_size):
-    draw.line(((i, 0), (i, frame.height)), fill=(0, 0, 0, 255))
-for i in range(0, frame.height, mv_block_size):
-    draw.line(((0, i), (frame.width, i)), fill=(0, 0, 0, 255))
+        rows, cols = struct.unpack('qq', contents[:16])
+        imps = np.frombuffer(contents[16:],
+                            dtype=np.float32).reshape((rows, cols))
 
-# Draw the importances.
-if max_imp > 0:
-    for y in range(rows):
-        for x in range(cols):
-            imp = imps[y, x]
-            top_left = (x * mv_block_size, y * mv_block_size)
-            bottom_right = (top_left[0] + mv_block_size, top_left[1] + mv_block_size)
-            draw.rectangle((top_left, bottom_right), fill=(int(imp / max_imp * 255), 0, 0, 128))
+        if raw:
+            click.secho("imps data after processing:", fg="red")
+            click.echo(imps)
+
+    elif path != None:
+        if verbose:
+            click.secho("MODE 2", fg="yellow")
+        png_list = glob.glob(str(path)+"/*hres.png")
+        bin_list = glob.glob(str(path)+"/*imps.bin")
```

To prevent errors I think it's best to get only one of these lists from a glob, and then string replace `imps` with `hres` or vice versa to get the second list. This way you will get an error upfront if you have a different amount of hres and imps files or if some of them are missing (rather than it silently using wrong files to produce wrong results).

---

Yes makes sense